src/main/java/org/elasticsearch/search/dfs/DfsPhase.java

```diff
@@ -61,7 +61,12 @@ public void execute(SearchContext context) {
                 context.updateRewriteQuery(context.searcher().rewrite(context.query()));
             }
 
-            context.query().extractTerms(new DelegateSet(termsSet));
```

Comment:
OK, let's fix BooleanQuery then. It is indeed the way how you can reach that point: a FilteredQuery against a QWF(TermsQuery) is rewritten to a BooleanQuery with the TermsQuery as a FILTER clause.